internal/command/volumes/create.go

```diff
@@ -79,6 +80,43 @@ func runCreate(ctx context.Context) error {
 		return err
 	}
 
+	// If the Yes flag has been supplied we skip the warning entirely, so no
+	// need to query for the set of volumes
+	if !flag.GetYes(ctx) {
+		// fetch the set of volumes for this app. If > 0 we skip the prompt
+		var volumes []api.Volume
+		if volumes, err = client.GetVolumes(ctx, appName); err != nil {
+			return err
+		}
+
+		var matches int32
+		for _, volume := range volumes {
+			if volume.Name == volumeName {
+				matches++
+			}
+		}
+
+		if matches == 0 {
+			var confirmed bool
+			io := iostreams.FromContext(ctx)
+			colorize := io.ColorScheme()
+
+			const msg = "Warning! Individual volumes are pinned to individual hosts. You should create two or more volumes per application. You will have downtime if you only create one. If you proceed, make sure to create a second volume with fly volume create --require-unique-zone to ensure a second volume exists on a separate worker"
+			fmt.Fprintln(io.ErrOut, colorize.Red(msg))
+
+			switch confirmed, err = prompt.Confirm(ctx, "Do you still want to use the volumes feature?"); {
+			case err == nil:
+				if !confirmed {
+					return nil
+				}
+			case prompt.IsNonInteractive(err):
+				return prompt.NonInteractiveError("yes flag must be specified when not running interactively")
+			default:
+				return err
+			}
+		}
+	}
```

Comment:
this function block is not part of the main flow and it is quite large, can we move that to another function that summarizes the intention instead of inlining the whole thing in runCreate?